src/GameplayKit/GKObstacleGraph.cs

```diff
@@ -0,0 +1,46 @@
+﻿//
+// GKObstacleGraph.cs: Implements Generic variant of GKObstacleGraph
+//
+// Authors:
+//	Alex Soto  <alexsoto@microsoft.com>
+//
+// Copyright 2016 Xamarin Inc. All rights reserved.
+//
+
+#if XAMCORE_2_0 || !MONOMAC
+
+using System;
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+
+namespace XamCore.GameplayKit {
+
+	[iOS (10,0), TV (10,0), Mac (10,12, onlyOn64: true)]
+	[Register ("GKObstacleGraph", SkipRegistration = true)]
+	public partial class GKObstacleGraph<NodeType> : GKObstacleGraph where NodeType : GKGraphNode2D {
+
+		[Preserve (Conditional = true)]
+		internal GKObstacleGraph (IntPtr handle) : base (handle)
+		{
+		}
+
+		public GKObstacleGraph (NSCoder coder) : base (coder)
+		{
+		}
+
+		public GKObstacleGraph (GKPolygonObstacle [] obstacles, float bufferRadius) : base (obstacles, bufferRadius, new Class (typeof (NodeType)))
+		{
+		}
+
+		public static GKObstacleGraph<NodeType> FromObstacles (GKPolygonObstacle [] obstacles, float bufferRadius, Class nodeClass)
```

Comment:
You mean hide by making it internal or just hide it from autocompletion? I would go for the later